bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/base/DefaultSampledSpectrum.java

```diff
@@ -54,6 +53,36 @@ public DefaultSampledSpectrum(ArraySpectrum sp, double xInterval) {
         reset(data, xStart, xInterval);
     }
 
+    /**
+     * Construct a DefaultSampledSpectrum by sampling the given ArraySpectrum
+     * over the specified wavelength range at the specified interval.
+     *
+     * @param sp        Spectrum to resample
+     * @param xStart    Starting wavelength of instrument configuration (nm)
+     * @param xEnd      Ending wavelength of instrument configuration (nm)
+     * @param xInterval Sampling interval (nm)
+     * @param z         Redshift of target
+     */
+    public DefaultSampledSpectrum(ArraySpectrum sp, double xStart, double xEnd, double xInterval, double z) {
+
+        if ( (1+z) * sp.getStart() > xStart || (1+z) * sp.getEnd() < xEnd ) {
+            throw new IllegalArgumentException(
+                    String.format("Redshifted SED (%.1f - %.1f nm) does not cover range of instrument configuration (%.1f - %.1f nm).",
+                            (1+z)*sp.getStart(), (1+z)*sp.getEnd(), xStart, xEnd));
+        }
+
+        // The SED will be redshifted later so resample over range / (1+z):
+        xStart /= 1+z;
```

Comment:
Should we check for z begin -1?

Reply:
That's a good question.  Large negative redshifts are not terribly physical (it implies movement *toward* us) so I'm not too worried if it fails.  Let me know if you think its worthwhile; it would be easy to throw an exception.